### ISSUE TITLE
if i changed specific clusterPoolIPv4PodCIDRList, node communiction fails

### ISSUE THREAD

**Issue by heeyoungEom-K** [cilium/cilium#32296]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When installing Cilium with Helm, deploying it with clusterPoolIPv4PodCIDRList set to 10.100.0.0/16 results in timeouts in communication with the node. with routingMode=tunnel
(Preconditions: kube-proxy removed, aws-node removed.)

```
helm install cilium cilium/cilium --version 1.15.4 --namespace kube-system --set egressMasqueradeInterfaces=eth0 --set kubeProxyReplacement=true --set k8sServiceHost=a.ap-northeast-2.eks.amazonaws.com --set k8sServicePort=443 --set cluster.id=1 --set cluster.name=eks --set ipam.operator.clusterPoolIPv4PodCIDRList=10.100.0.0/16 --set routingMode=tunnel 
```

issue
- cilium timeout, image pull failed
- coredns timeout, imgae pull failed
- other deploymenet pod timeout, imgae pull failed

try it
- change node
- change eks

![image](https://github.com/cilium/cilium/assets/161670432/afc0f9aa-d73a-452e-9c87-533a6f11b591)

### Cilium Version

1.15.4

### Kernel Version

5.10.213-201.855.amzn2.x86_64

### Kubernetes Version

1.29.0

### Regression

-

### Sysdump

Collecting tracing data from Cilium pods
🔍 Collect Kubernetes nodes
🔍 Collecting Kubernetes events
🔍 Collect Kubernetes version
🔍 Collecting Kubernetes pods
🔍 Collecting Kubernetes namespaces
🔍 Collecting Kubernetes services
🔍 Collecting Kubernetes pods summary
🔍 Collecting Kubernetes endpoints
🔍 Collecting Kubernetes metrics
🔍 Collecting Kubernetes network policies
🔍 Collecting Kubernetes leases
🔍 Collecting Cilium network policies
🔍 Collecting Cilium Egress Gateway policies
🔍 Collecting Cilium egress NAT policies
🔍 Collecting Cilium cluster-wide network policies
🔍 Collecting Cilium CIDR Groups
🔍 Collecting Cilium endpoint slices
🔍 Collecting Cilium endpoints
🔍 Collecting Cilium nodes
🔍 Collecting Cilium identities
🔍 Collecting Ingresses
🔍 Collecting Cilium Node Configs
🔍 Collecting Cilium BGP Peering Policies
🔍 Collecting IngressClasses
🔍 Collecting Cilium LoadBalancer IP Pools
🔍 Collecting Cilium local redirect policies
🔍 Collecting Cilium Pod IP Pools
🔍 Checking if cilium-etcd-secrets exists in kube-system namespace
🔍 Collecting the Cilium configuration
🔍 Collecting the Cilium Node Init daemonset
🔍 Collecting the Cilium daemonset(s)
🔍 Collecting the Cilium Envoy configuration
🔍 Collecting the Cilium Envoy daemonset
🔍 Collecting the Hubble daemonset
🔍 Collecting the Hubble Relay configuration
🔍 Collecting the Hubble Relay deployment
🔍 Collecting the Hubble UI deployment
🔍 Collecting the Cilium operator deployment
🔍 Collecting the Cilium operator metrics
🔍 Collecting the clustermesh metrics
🔍 Collecting the 'clustermesh-apiserver' deployment
🔍 Collecting the CNI configuration files from Cilium pods
🔍 Collecting the CNI configmap
🔍 Collecting gops stats from Cilium pods
🔍 Collecting gops stats from Hubble pods
⚠️ Daemonset "cilium-envoy" not found in namespace "kube-system" - this is expected if Envoy DaemonSet is not enabled
⚠️ Deployment "hubble-relay" not found in namespace "kube-system" - this is expected if Hubble is not enabled
🔍 Collecting gops stats from Hubble Relay pods
🔍 Collecting bugtool output from Cilium pods
⚠️ Deployment "hubble-ui" not found in namespace "kube-system" - this is expected if Hubble UI is not enabled
🔍 Collecting profiling data from Cilium pods
⚠️ Deployment "clustermesh-apiserver" not found in namespace "kube-system" - this is expected if 'clustermesh-apiserver' isn't enabled
🔍 Collecting logs from Cilium pods
⚠️ Container "cilium-agent" for pod "cilium-c6nl7" in namespace "kube-system" is not running. Trying EphemeralContainer or separate Pod instead...
🔍 Collecting logs from Cilium Envoy pods
Secret "cilium-etcd-secrets" not found in namespace "kube-system" - this is expected when using the CRD KVStore
I0502 11:01:02.916569   25063 request.go:697] Waited for 1.057614458s due to client-side throttling, not priority and fairness, request: GET:https://8366EA28884933BF0A504422D1B5E370.yl4.ap-northeast-2.eks.amazonaws.com/api/v1/namespaces/kube-system/configmaps/hubble-relay-config
🔍 Collecting logs from Cilium Node Init pods
🔍 Collecting logs from Cilium operator pods
🔍 Collecting logs from 'clustermesh-apiserver' pods
🔍 Collecting logs from Hubble pods
🔍 Collecting logs from Hubble Relay pods
🔍 Collecting logs from Hubble UI pods
🔍 Collecting platform-specific data
🔍 Collecting kvstore data
🔍 Collecting Cilium external workloads
🔍 Collecting Hubble flows from Cilium pods
🔍 Collecting logs from Tetragon pods
🔍 Collecting logs from Tetragon operator pods
🔍 Collecting bugtool output from Tetragon pods
🔍 Collecting Tetragon configmap
🔍 Collecting Tetragon PodInfo custom resources
🔍 Collecting Tetragon tracing policies
🔍 Collecting Tetragon namespaced tracing policies
🔍 Collecting Helm values from the release
⚠️ EphemeralContainer "sysdump-1714615262" on pod "" in namespace "" never reached Running status (falling back to separate Pod)
⚠️ Container "cilium-agent" on pod "cilium-c6nl7" in namespace "kube-system" is not running. Creating exec Pod.

### Relevant log output

```shell
-
```

### Anything else?

-

### Cilium Users Document

- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by youngnick** [comment]
Thanks for this issue @heeyoungEom-K, the sysdump process should have provided a zip file, someone will need that to be able to take a look.

Also, could you clarify your environment? Are you running on EKS? Is there any way that subnet (`10.100.0.0/16`) is also in use somewhere else? If it was, then things could get weird.

**Comment by heeyoungEom-K** [comment]
> Thanks for this issue @heeyoungEom-K, the sysdump process should have provided a zip file, someone will need that to be able to take a look.
> 
> Also, could you clarify your environment? Are you running on EKS? Is there any way that subnet (`10.100.0.0/16`) is also in use somewhere else? If it was, then things could get weird.

hello @youngnick 
thanks for your replay,
i can't share my Sysdump file because the cilium-operator had broken down and stop logging the sysdump
I am not using it(`10.100.0.0/16`) in other EKS

VPC CIDR: 10.100.0.0/16
Subnet: 10.100.0.0/20, 10.100.16.0/20 (public), 10.100.128.0/20, 10.100.144.0/20 (private)
Service CIDR: 10.110.0.0/16"

**Comment by youngnick** [comment]
If you're installing with routing mode = "tunnel", then you can't use the same CIDR as the VPC CIDR (`10.100.0.0/16`) for the internal cluster network (which you said in the first post you were setting to `10.100.0.0/16` with `clusterPolIPv4PodCIDRList). If you do, then the overlay network and the VPC network would be using the same IP addresses, and things will fail in the way you described, since the VPC network will not be routable, which means that traffic to outside the cluster will fail.

As you say, I would recommend just not setting that value.

**Comment by heeyoungEom-K** [comment]
> If you're installing with routing mode = "tunnel", then you can't use the same CIDR as the VPC CIDR (`10.100.0.0/16`) for the internal cluster network (which you said in the first post you were setting to `10.100.0.0/16` with `clusterPolIPv4PodCIDRList). If you do, then the overlay network and the VPC network would be using the same IP addresses, and things will fail in the way you described, since the VPC network will not be routable, which means that traffic to outside the cluster will fail.
> 
> As you say, I would recommend just not setting that value.

thank you for replay @youngnick 

I am not well-versed in Cilium or network configurations, so my settings might be inappropriate. I am in a situation where I need to connect ClusterMesh between Azure and AWS. AKS defaults to a BYOCNI (Encapsulation) mode set to cluster pool, so I have set EKS in tunnel mode instead of Direct Routing mode. (As far as I know, only identical data paths can communicate with each other.)

I set the IP using ClusterPoolIPv4PodCIDRList because I wanted to connect the clusters via a VPN (site-to-site, BGP) to route traffic to specific pods.

Is there any unnecessary content in my setup or cases where I could bypass certain settings?

**Comment by youngnick** [comment]
Sorry, @heeyoungEom-K, I'm only triaging issues this week and I can't help you with that. I would encourage you to look for some further guides or assistance only for getting your EKS install working in VXLAN/tunnel mode so that you can start on the clustermesh part (and I'd read https://docs.cilium.io/en/stable/network/clustermesh/ closely as well).

**Comment by heeyoungEom-K** [comment]
> clusterPoolIPv4PodCIDRList

don't mind thank you for replay @youngnick :)

I've completed clustermesh connection tests(aks-to-eks). 
I was wondering why eks would break if I registered the clusterPoolIPv4PodCIDRLlist.

**Comment by youngnick** [comment]
As I said, that same range is shared between the outer VPC network, and then reused in the tunnel network (which is a software-defined network that runs across all the nodes). That setup won't work, because traffic needs to be able to pass through the outer VPC network to get to other networks (like yours or the internet), and if the tunnel network thinks it _is_ that network, then it won't route to the outer network.

The short version is "your cluster network needs to be in a _separate_ range than your VPC network when you are doing tunnelling mode".

If you got the clustermesh working, does that answer your question?

**Comment by heeyoungEom-K** [comment]
> clusterPoolIPv4PodCIDRList

Yes, thank you for your answer. I will close this issue here.
